Modules/_testcapimodule.c

```diff
@@ -3231,6 +3231,86 @@ test_weakref_capi(PyObject *Py_UNUSED(module), PyObject *Py_UNUSED(args))
 }
 
 
+static PyObject *
+sys_getattr(PyObject *Py_UNUSED(module), PyObject *name)
+{
+    NULLABLE(name);
+    PyObject *result = PySys_GetAttr(name);
+    if (result == NULL && PyErr_Occurred()) {
+        return NULL;
+    }
+    if (result == NULL) {
+        result = PyExc_AttributeError;
+        Py_INCREF(PyExc_AttributeError);
+    }
+    return result;
+}
+
+static PyObject *
+sys_getattrstring(PyObject *Py_UNUSED(module), PyObject *arg)
+{
+    const char *name;
+    Py_ssize_t size;
+    if (!PyArg_Parse(arg, "z#", &name, &size)) {
+        return NULL;
+    }
+    PyObject *result = PySys_GetAttrString(name);
+    if (result == NULL && PyErr_Occurred()) {
+        return NULL;
+    }
+    if (result == NULL) {
+        result = PyExc_AttributeError;
+        Py_INCREF(PyExc_AttributeError);
+    }
+    return result;
+}
+
+static PyObject *
+sys_getoptionalattr(PyObject *Py_UNUSED(module), PyObject *name)
+{
+    PyObject *value = UNINITIALIZED_PTR;
+    NULLABLE(name);
+
+    switch (PySys_GetOptionalAttr(name, &value)) {
+        case -1:
+            assert(value == NULL);
```

Comment:
I suggest adding: ``assert(PyErr_Occurred());``. Same remark in sys_getoptionalattrstring().